packages/core/src/util/util.ts

```diff
@@ -77,7 +77,17 @@ export const deriveTypes = (jsonSchema: JsonSchema): string[] => {
   if (!isEmpty(jsonSchema.items)) {
     return ['array'];
   }
-
+  if (!isEmpty(jsonSchema.enum)) {
+    const types: Set<string> = new Set();
+    jsonSchema.enum.forEach((enumElement) => {
+      if (typeof enumElement === 'string') {
+        types.add('string');
+      } else {
+        deriveTypes(enumElement).forEach((type) => types.add(type));
```

Comment:
We need to check whether the array has a single item. If yes, then we should return that directly instead of an array

Reply:
If we want to do that we should adapt the type definition of the deriveTypes function as it only declares string[] as a return type. Furthermore, we also return single value arrays further above. So IMO we can leave it like that.

Reply:
@sdirix Or am I missing something?

Reply:
@lucas-koehler you are right. I was too fast when reviewing and got the use case of this function confused. The code is fine.